cmd/tapcli/universe.go

```diff
@@ -698,6 +715,218 @@ func universeFederationDel(ctx *cli.Context) error {
 	return nil
 }
 
+var universeFederationConfigCommand = cli.Command{
+	Name:      "config",
+	ShortName: "c",
+	Usage:     "Change the sync behavior of the local Universe",
+	Description: `
+	Manage the sync behavior of the local Universe. These settings are
+	defined by the proof type (issuance or transfer), the sync behavior
+	(insert from remote Universe or export to remote Universe), and the
+	scope (all assets or specific assets).
+        `,
+	Subcommands: []cli.Command{
+		universeFederationGlobalConfig,
+		universeFederationLocalConfig,
+	},
+}
+
+var (
+	proofInsertName        = "insert"
+	proofExportName        = "export"
+	universeSyncConfigArgs = []cli.Flag{
+		cli.StringFlag{
+			Name:  proofTypeName,
+			Usage: "the type of proof",
+		},
+		cli.StringFlag{
+			Name: proofInsertName,
+			Usage: "if true, remote Universes can push proofs to" +
+				"the local Universe",
+		},
+		cli.StringFlag{
+			Name: proofExportName,
+			Usage: "if true, remote Universes can pull proofs from" +
+				"the local Universe",
+		},
+	}
+)
+
+func parseConfigArgs(ctx *cli.Context,
+	isGlobal bool) (*unirpc.GlobalFederationSyncConfig,
+	*unirpc.AssetFederationSyncConfig, error) {
+
+	var (
+		insertOpt *bool
+		exportOpt *bool
+		err       error
+	)
+
+	// The cli.BoolFlag works for variables that are false by default,
+	// but here we don't want a default for these values.
+	isValidBool := func(arg string) (bool, error) {
+		switch arg {
+		case "true":
+			return true, nil
+		case "false":
+			return false, nil
+		default:
+			return false, fmt.Errorf("invalid bool flag")
+		}
+	}
+
+	if !ctx.IsSet(proofInsertName) && !ctx.IsSet(proofExportName) {
+		return nil, nil, fmt.Errorf("either insert or export must be" +
+			"set")
+	}
+
+	// The insert and export flags are not mutually exclusive.
+	if ctx.IsSet(proofInsertName) {
+		*insertOpt, err = isValidBool(ctx.String(proofInsertName))
+		if err != nil {
+			return nil, nil, fmt.Errorf("invalid proof insert flag")
+		}
+	}
+
+	if ctx.IsSet(proofExportName) {
+		*exportOpt, err = isValidBool(ctx.String(proofExportName))
```

Comment:
Unclear why we need special proof handling here.

Reply:
Not sure what you meant?

Reply:
Nvm, makes sense re decision to force a user to specify everything.